lightning/restore/restore.go

```diff
@@ -908,13 +908,17 @@ func extractTiDBVersion(version string) (*semver.Version, error) {
 	//                               ^~~~^
 	// version format: "5.7.10-TiDB-v2.0.7"
 	//                               ^~~~^
+	// version format: "5.7.25-TiDB-v3.0.0-beta-211-g09beefbe0-dirty"
+	//                               ^~~~^
 	// The version is generated by `git describe --tags` on the TiDB repository.
 	versions := strings.Split(version, "-")
 	end := len(versions)
 	switch end {
 	case 3, 4:
 	case 5, 6:
 		end -= 2
+	case 7:
+		end -= 3
```

Comment:
is the version format fixed?

Reply:
I think what can be determined is `5.x.x-TiDB-{tidb version}-xxxxxxx`, it's better to extract front three item

Reply:
The version format was generated by `git describe --tags --dirty --always`, so I think it will be a fixed format.